sklearn/decomposition/_online_lda_fast.pyx

```diff
@@ -8,8 +10,17 @@ from libc.math cimport exp, fabs, log
 from numpy.math cimport EULER
 
 
-def mean_change(np.ndarray[ndim=1, dtype=np.float64_t] arr_1,
-                np.ndarray[ndim=1, dtype=np.float64_t] arr_2):
+# floating2 is introduced in order to accept two different
+# dtypes for mean_change and _dirichlet_expectation_1d methods.
+# By accepting two different two dtypes, unnecessary memory allocation is
+# suppressed along with type conversion in prediction time.
```

Comment:
Does this reads better?
```suggestion
# floating2 is introduced in order to accept parameters with two different
# dtypes for `mean_change` and `_dirichlet_expectation_1d`.
# By accepting two different two dtypes, unnecessary memory allocation is
# suppressed along with type conversion in prediction time.
```

What do you mean by "unnecessary memory allocation is suppressed"?

Reply:
> What do you mean by "unnecessary memory allocation is suppressed"?
If we should pass two arrays with same dtype, we should convert dtype of either of arrays in python code before passing the two arrays to the functions. The type conversion require extra buffer for the converted array.